src/compute/src/arrangement/manager.rs

```diff
@@ -98,77 +97,8 @@ impl TraceManager {
     }
 
     /// Removes the trace for `id`.
-    pub fn del_trace(&mut self, id: &GlobalId) -> bool {
-        self.traces.remove(id).is_some()
-    }
-}
-
-/// An abstraction of a trace handle.
-///
-/// This type exists as an `enum` to support potential experimentation with alternate
-/// representation and layouts.
-#[derive(Clone)]
-pub enum SpecializedTraceHandle {
```

Comment:
I'm not entirely clear whether removing `SpecializedTraceHandle` is the right way forwards. Ideally, we have _some_ mechanism that allows us to have different kinds of arrangements and present them in a similar way outwards, and `SpecializedTraceHandle` gives us this behavior. However, I can understand that it's in the way. How much effort is it not to rip it out?

Reply:
> How much effort is it not to rip it out?

Took me a while to find an approach that works, but here you go: https://github.com/MaterializeInc/materialize/pull/27593